### PR TITLE
All cameras now handle the "up" property correctly

### PR DESCRIPTION
**Pull request by almarklein**
PanZoom can thus operate in any plane, Turntable can be tilted in any plane, and FlyCamera will attempt to re-orient to make up point upward.

### CODE REVIEW

File: vispy/scene/cameras/cameras.py [hunk 1]
```diff
@@ -688,10 +689,42 @@ def _update_transform(self):
...
+        # transform When up is +z this is all. In other cases,
+        # self.transform is now set up correctly to allow pan/zoom, but
+        # for the scene we need a different (3D) mapping. When there
+        # is a minus in up, we simply look at the scene from the other
+        # side (as if z was flipped).
+        
+        if self.up == '+z':
+            # Keep it simple, but we give the scene an affinetransform
+            # to be consistent
+            thetransform = self.transform * AffineTransform()
```
Comment (campagnola) [inline vispy/scene/cameras/cameras.py]: I'd rather it left as-is because affine should be a bit slower.

Comment (Eric89GXL) [inline vispy/scene/cameras/cameras.py]: Doesn't it end up as a dot product on the GPU either way? Or do you mean in the scenegraph before it gets to the GPU?

Comment (campagnola) [inline vispy/scene/cameras/cameras.py]: Most transforms are just scale+translate.

Comment (almarklein) [inline vispy/scene/cameras/cameras.py]: I had it unmodified at first, but then when I cycled through the up vectors, the z+ would not work, maybe a problem with transform caching? Anyway, I cannot imagine that the performance difference is noticeable.

Comment (Eric89GXL) [inline vispy/scene/cameras/cameras.py]: Yeah, but I thought these were done on GPU with matrix multiplies either way...

**Pull request merged**